real-main/app/models/user/dynamo/base.py

```diff
@@ -168,6 +173,16 @@ def process_attr(name, value):
                 else:
                     expression_actions['REMOVE'].append(name)
 
+        if current_location is not None:
+            latitude = BasicContext.create_decimal(current_location['latitude']).normalize()
+            longitude = BasicContext.create_decimal(current_location['longitude']).normalize()
+            accuracy = current_location.get('accuracy')
+
+            if accuracy is not None:
+                accuracy = BasicContext.create_decimal(accuracy).normalize()
+
+            current_location = {"latitude": latitude, "longitude": longitude, "accuracy": accuracy}
```

Comment:
```suggestion
        for key in ('latitude', 'longitude'):
            current_location[key] = BasicContext.create_decimal(current_location[key]).normalize()
```

We actually don't need to apply the conversion to `Decimal` to accuracy because it is an integer, not a float.

Reply:
You may need to check and handle the situation where `accuracy` is in `current_location`, but has value `None`. I'm not sure if this is needed - it depends on how appsync/graphql handles the situation where `accuracy`:
  - is just missing from the GQL type
  - is in the GQL type with value `null`

So you might need to add something like this:

```py
if current_location.get('accuracy') is None:
  current_location.pop('accuracy', None)
```